Backend/app/dashboard/CommonAPIs/src/Dashboard/RiderPlatform/Ride.hs

```diff
@@ -81,3 +87,55 @@ data LocationAddress = LocationAddress
     area :: Maybe Text
   }
   deriving (Generic, Show, Eq, ToSchema, FromJSON, ToJSON)
+
+---------------------------------------------------------
+-- ride list --------------------------------------------
+
+type RideListAPI =
+  "list"
+    :> QueryParam "limit" Int
+    :> QueryParam "offset" Int
+    :> QueryParam "bookingStatus" BookingStatus
+    :> QueryParam "rideShortId" (ShortId Ride)
+    :> QueryParam "customerPhoneNo" Text
+    :> QueryParam "driverPhoneNo" Text
+    :> Get '[JSON] RideListRes
+
+data RideListRes = RideListRes
+  { totalItems :: Int, -- for backward compatibility
+    summary :: Summary,
+    rides :: [RideListItem]
+  }
+  deriving stock (Show, Generic)
+  deriving anyclass (ToJSON, FromJSON, ToSchema)
+
+data RideListItem = RideListItem
+  { rideShortId :: ShortId Ride,
+    rideCreatedAt :: UTCTime,
+    rideId :: Id Ride,
+    customerName :: Maybe Text,
+    customerPhoneNo :: Maybe Text,
+    driverName :: Text,
+    driverPhoneNo :: Text,
+    vehicleNo :: Text,
+    bookingStatus :: BookingStatus
+  }
+  deriving stock (Show, Generic)
+  deriving anyclass (ToJSON, FromJSON, ToSchema)
+
+data BookingStatus = UPCOMING | UPCOMING_6HRS | ONGOING | ONGOING_6HRS | RCOMPLETED | RCANCELLED
```

Comment:
this type can be reused from `ProviderPlatform`

Reply:
In Booking Status I used RCOMPLETED and RCANCELLED at `riderPlatform` , at `ProviderPlatform` we have COMPLETED and CANCELLED . If i try to use these two COMPLETED and CANCELLED at `riderPlatform` side also then I'm facing error because these are defined in Ride Status.